flickit-assessment-data/src/main/java/org/flickit/assessment/data/jpa/core/assessmentinsight/AssessmentInsightJpaEntity.java

```diff
@@ -0,0 +1,35 @@
+package org.flickit.assessment.data.jpa.core.assessmentinsight;
+
+import jakarta.persistence.*;
+import lombok.*;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Entity
+@Table(name = "fac_assessment_insight")
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+public class AssessmentInsightJpaEntity {
+
+    @Id
+    @EqualsAndHashCode.Include
+    @GeneratedValue(strategy = GenerationType.AUTO)
```

Comment:
چرا 
`@GeneratedValue(strategy = GenerationType.UUID)`
نذاشتیم؟
اینجور تغییرات رو اگر خودتون کامنت بذارید و توضیح بدید که چرا متفاوت با قبل عمل کردیم، دیگه من برام سوال نمی‌شه و رفت و برگشت‌مون کمتر می‌شه.
انوتیشن `GenericGenerator` هم حذف شده.
اگر این تغییرات نتیجه بررسی‌ای چیزی بوده، ممنون می‌شم اطلاع بدید که تیم هم در جریان قرار بگیره و تصمیم بگیریم که از این به بعد اینطوری باید پیش بریم یا نه

Reply:
من یادم رفت annotation رو بهش اضافه کنم.
مسئله این هست که strategy توی hibernate نسخه ۶.۲ به بعد deprecate شده و به جاش از `@UuidGenerator(style = UuidGenerator.Style.TIME)` استفاده می کنند البته style-های مختلفی وجود داره و time کارایی بالاتری داره، استایل پیش فرض که از randomness استفاده می کنه خیلی کارایی رو میاره پایین، متاسفانه الان همه جا از randomness استفاده کردیم. یه موردی که هست اینه که  postgres به صورت پیش فرض روی کلید اصلی ایندکس ایجاد می کنه و کلیدهای uuid که براساس randomness تولید میشن خیلی کارایی رو میارن پایین. و خب استایل time برای بهبود این قضیه معرفی شده. این دو مقاله رو یه نگاه اجمالی کنید لطفا.
[مقاله ۱](https://thorben-janssen.com/generate-uuids-primary-keys-hibernate/#:~:text=Summary-,Generating%20UUIDs%20using%20JPA%203.1,based%20on%20IETF%20RFC%204122.&text=Let's%20try%20this%20mapping%20and%20persist%20a%20new%20Book%20entity%20object.)
[مقاله ۲](https://dev.to/mcadariu/using-uuids-as-primary-keys-3e7a)

Reply:
دستتون درد نکنه
اول اینکه به نظرتون پارامتر columnDefinition = "uuid" ضرورت داره تعریفش؟ آیا مصداق database-specific configuration نیست؟ ما نمی‌خوایم هیچ چیزی که ممکنه خاص Postgress باشه تو لایه دیتا داشته باشیم. برای کوئری‌ها هم خیلی حواسمون که مثلا از native query جایی استفاده نکنیم. اگر به نظرتون اوکیه و این مساله‌ای که گفتم پیش نمیاد، نگهش داریم. وگرنه حذفش کنیم.

در مورد رندوم و تایم هم هر دو مقاله رو خوندم و چندتا نکته و دغدغه دارم:

1. من این برداشت رو نکردم که randomness کارایی رو «خیلی» میاره پایین
2. خود جاواداک روی TIME یه نیمچه هشداری روی پیاده‌سازی داده که البته من نرفتم پیاده‌سازی هایبرنیت رو بررسی کنم دیگه:
> Implementation Note:
> Can be a bottleneck, since synchronization is used when incrementing an internal counter as part of the algorithm.

3. همچین چیزی هم یه جا دیدم:
> While time-based UUIDs are often more performant in distributed systems, they may expose time-related information, which could be a security or privacy concern in some contexts.

4. تو اسکیلی که ما داریم و خواهیم داشت، بعید می‌دونم رندومنس مساله‌ای ایجاد کنه برامون.

خلاصه به اندازه شما مطمئن نیستم ولی مخالفت خاصی هم ندارم.
می‌تونیم برای انتیتی‌های جدید از این به بعد اینطوری پیش بریم.
قبلی‌ها رو هم اگر مطمئن بودیم مشکلی پیش نمیاد، از رندوم به تایم تغییر بدیم بعدا.

شما فقط همون columnDefinition رو جواب بدید، بعدش مرج می‌شه این برنچ

Reply:
آهان یه چیز کلی هم بگم. الان که نسخه جاوا رو بردیم بالا، مجبور نیستیم استایل کد یا روش‌هامون رو هم بلافاصله تغییر بدیم. منظورم این تسک‌ نبودا. کلی گفتم که برای تسک‌های بعدی در نظر داشته باشید.
همه چیز رو طبق قبل ادامه میدیم و هر جا روش متفاوت یا بهبود خاصی مد نظرمون بود، تو لیست ریفکتور میاریم.

Reply:
- اگر در scale ما تاثیر گذار نیست که من هم موافقم که به همون صورتی که الان تعریف کردیم بمونه.(یعنی random)
- در رابطه با column definition، به نظرم حذفش کنیم بهتره
فقط الان متوجه نشدم، از انوتیشن جدید استفاده نکنیم فعلا؟ یعنی با همون اتریبیوت strategy زده بشه؟

Reply:
حالا به نظرم می‌تونیم پرونده‌ش رو نبدیم و تا داده‌هامون زیاد نشده، وقت داریم تغییرش بدیم.

اگر بخوایم با رندوم پیش بریم، این دو حالت جفتش درسته؟

```java
@Id
@EqualsAndHashCode.Include
@GeneratedValue
@UuidGenerator
@Column(name = "id", updatable = false, nullable = false)
private UUID id;

@Id
@EqualsAndHashCode.Include
@GeneratedValue(strategy = GenerationType.UUID)
@Column(name = "id", updatable = false, nullable = false)
private UUID id;
```

من برام فرقی نداره. هر کدوم که خواستید استفاده کنید.
یه پیام هم تو گروه بذاریم که از این به بعد بچه‌ها این کار رو کنن.

Reply:
باتوجه به اینکه strategy توی نسخه هایبرنیت ۶.۲ deprecate شده، به نظرم بهتره که به صورت زیر پیش بریم:

```java
@Id
@EqualsAndHashCode.Include
@GeneratedValue
@UuidGenerator(style = UuidGenerator.Style.RANDOM)
@Column(name = "id", updatable = false, nullable = false)
private UUID id;
```